packages/protocol/test/bridge/Bridge.test.ts

```diff
@@ -1010,11 +1014,675 @@ describe("integration:Bridge", function () {
+                ["bytes", "bytes"],
+                [
+                    RLP.encode(proof.accountProof),
+                    RLP.encode(proof.storageProof[0].proof),
+                ]
+            )
+            // encode the SignalProof struct from LibBridgeSignal
+            const signalProof = ethers.utils.defaultAbiCoder.encode(
+                [
+                    "tuple(tuple(bytes32 parentHash, bytes32 ommersHash, address beneficiary, bytes32 stateRoot, bytes32 transactionsRoot, bytes32 receiptsRoot, bytes32[8] logsBloom, uint256 difficulty, uint128 height, uint64 gasLimit, uint64 gasUsed, uint64 timestamp, bytes extraData, bytes32 mixHash, uint64 nonce, uint256 baseFeePerGas) header, bytes proof)",
+                ],
+                [{ header: blockHeader, proof: encodedProof }]
+            )
+            await expect(
+                l2Bridge.isMessageReceived(signal, srcChainId, signalProof)
+            ).to.be.revertedWith("Invalid large internal hash")
+        })
+
+        it("should return true", async function () {
+            const {
+                owner,
+                l1Bridge,
+                srcChainId,
+                enabledDestChainId,
+                l2Bridge,
+                headerSync,
+            } = await deployBridgeFixture()
+
+            const m: Message = {
+                id: 1,
+                sender: owner.address,
+                srcChainId: srcChainId,
+                destChainId: enabledDestChainId,
+                owner: owner.address,
+                to: owner.address,
+                refundAddress: owner.address,
+                depositValue: 1000,
+                callValue: 1000,
+                processingFee: 1000,
+                gasLimit: 10000,
+                data: ethers.constants.HashZero,
+                memo: "",
+            }
+
+            const expectedAmount =
+                m.depositValue + m.callValue + m.processingFee
+            const tx = await l1Bridge.sendMessage(m, {
+                value: expectedAmount,
+            })
+
+            const receipt = await tx.wait()
+
+            const [messageSentEvent] = receipt.events as any as Event[]
+
+            const { signal } = (messageSentEvent as any).args
+
+            const sender = l1Bridge.address
+
+            const key = ethers.utils.keccak256(
+                ethers.utils.solidityPack(
+                    ["address", "bytes32"],
+                    [sender, signal]
+                )
+            )
+
+            // use this instead of ethers.provider.getBlock() beccause it doesnt have stateRoot
+            // in the response
+            const block: Block = await ethers.provider.send(
+                "eth_getBlockByNumber",
+                ["latest", false]
+            )
+
+            await headerSync.setSyncedHeader(block.hash)
+
+            const logsBloom = block.logsBloom.toString().substring(2)
+
+            const blockHeader: BlockHeader = {
```

Comment:
we re-use this `get latest block and then convert logsbloom and then convert to blockheader` logic a bit, can it be in a utils function? Would remove a lot of lines from this file.